### ISSUE TITLE
[Bug]: select for update which return null data reported panic error

### ISSUE THREAD

**Issue by heni02** [matrixorigin/matrixone#10298]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Environment

```Markdown
- Version or commit-id (e.g. v0.1.0 or 8b23a93):84df60c4081376ddcf184a2596121487911696f4
- Hardware parameters:
- OS type:
- Others:
```

### Actual Behavior
![image](https://github.com/matrixorigin/matrixone/assets/113406637/6e1877b4-70c5-4b96-818a-de81d06b6aff)

### Steps to Reproduce

```Markdown
create table su_01(c1 int not null,c2 varchar(25),c3 int,primary key(c1),unique index u1(c2));
insert into su_01 values(1,'results',20),(2,'plo',50),(3,'kelly',60),(4,'yellow',70);

begin;
select * from su_01 where c1=7 for update;
```

**Comment by heni02** [comment]
@zhangxu19830126 plz confirm this issue

**Comment by zhangxu19830126** [comment]
fixed by #10375 10398

**Comment by heni02** [comment]
confirm,closed
commit id:48a289863740d954b9e48fbc889fc100a1f85198
<img width="1040" alt="企业微信截图_c9a66f78-4014-4d01-8de1-75dc4c9c4512" src="https://github.com/matrixorigin/matrixone/assets/113406637/71005794-7ba2-48a9-971f-d6b902d62fd1">
